### PR TITLE
Adds support for 3DTILES_content_gltf and EXT_feature_metadata

### PR DESCRIPTION
**Pull request by sanjeetsuhag**
This PR adds support for 3DTILES_content_gltf and EXT_feature_metadata to target 3D Tiles Next.

### CODE REVIEW

File: CDBTo3DTiles/src/CDBTo3DTiles.cpp [hunk 1]
```diff
@@ -86,6 +87,12 @@ struct Converter::Impl
                               const std::filesystem::path &outputDirectory,
                               CDBTileset &tilesetCollections);
 
+    void createGLTFForTileset(tinygltf::Model &model,
+                              CDBTile cdbTile,
+                              const CDBInstancesAttributes *instancesAttribs,
+                              const std::filesystem::path &outputDirectory,
+                              CDBTileset &tilesetCollections);
```
Comment (ErixenCruz) [inline CDBTo3DTiles/src/CDBTo3DTiles.cpp:94]: We will need to add comments explaining functions and all parameters they take?

File: CDBTo3DTiles/src/TileFormatIO.cpp [hunk 2]
```diff
@@ -361,6 +393,159 @@ void createBatchTable(const CDBInstancesAttributes *instancesAttribs,
...
+            // Add data to buffer
+            tinygltf::Buffer buffer;
+            auto &bufferData = buffer.data;
+            bufferData.resize(propertyBufferLength);
+            std::memcpy(bufferData.data(), property.second.data(), propertyBufferLength);
+            gltf->buffers.emplace_back(buffer);
+
+            // Add buffer view for property
+            tinygltf::BufferView bufferView;
+            bufferView.buffer = static_cast<int>(gltf->buffers.size() - 1);
```
Comment (ErixenCruz) [inline CDBTo3DTiles/src/TileFormatIO.cpp:430]: why is this size-1 instead of size?

File: CDBTo3DTiles/src/TileFormatIO.h [hunk 3]
```diff
@@ -6,6 +6,7 @@
 #include <filesystem>
 #include <fstream>
 #include <functional>
+#include <iostream>
```
Comment (ErixenCruz) [inline CDBTo3DTiles/src/TileFormatIO.h:9]: where is iostream used? was it left here after debugging with some couts?

**Review by ErixenCruz — COMMENTED** [review]
What commands are you using to test on san diego and yemen, and what changes are you making to those datasets for testing specifically? i want to reproduce exactly as you

**Comment by sanjeetsuhag** [comment]
Instructions for testing:

- Change the name of all geocells/datasets that should not be tested - for this PR, only keep `GSModel` and `Vector` datasets. 
image
- Use the [feature-metadata](https://github.com/CesiumGS/cesium/tree/feature-metadata) branch of CesiumJS. Make sure to pull in latest changes from `master` on to your local branch.
- Use this [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=7Vptb9s4Ev4rXH+JgyaUc90DiiQOLnG8jbHOC9buFos4MGSJsbilRUGk7HgP/e83pEi9WYrttCiwi/vQxKZmOO/PzKhxHHSJBF1EjCCfLDjiz+j9NRpTRgR6Jq5MYoIi6n2h4RytqAxQwJckRm7oI0EY8SSakcBdUh5PQsdBVwllvqL1XekijyexJGKtbr37o4fuIxJeqwcRj6XLTlEgZXTqOKvV6gSHaw/P+dIRVBLH51RKh4ZUUlfSJRHOe/94Zi7HkTsnk3DpxmhJyQrU6aKQrFCPCJos8O/6rD1pefp7j4fSpSGJJ60j9N9JiJAkcQwnDzFfUp/Ep5bRi8Fg8pnHzB+nJO3Do0n49fBsEoI4fS0WHgkJnjM+I9gnkQzGRMjLORALabhAHRknRHNprwy56yMZEO0Ea4VAUnmZyNQS86VsSvrr/fU4fdjW6icxeO7AuI5xz2UBF/L0Q+fDiaMJnbv3/3I+n5x8cD6ObrlPmHBOpifqVB1Or993OtNRp3MyHcMP/Kfg4YE1s2RkFNMF1d7Hru+3jYaKzHzE4DB/DX5cUEEwGBi2n5PQk5SHqH2YOtvc+BfnizFvqxNkbT1KvxXsvYH7wDUPVHrBb244J4YBoQ7uHNnPxx387+yLVWXGk1DxjqKAxATHcFEikIN+xp2U9FD9UsqbeEJcbsa3Q6Tymblr9MxjJAK+UtlrMz90FwSBMQueCKII01ip43vD1kU+95IFCaXJnz4j6hvkn0+Xk1bBq57NROxGUAl+L4BMaBcuU7SFr9hjrhB3SocumrRmrvfFj3k0aVXIhFwzSEYqolShSSvkIWkgizgUmAqQonNngrNENtHOuJR8oSk7DSSMPMvXCB4nrYjTEErumCzBLWLSetqqImSbioIi+zl6aVIO3DGPVdR7nPE4dRGDQ02v4zsIIagLV5vrQoJIXYReEsegCVsbBCO+jXca3ey0m2aweXiKQBR5hvj5Ovl4TOc0dJmWflqqWnWSQofV5DJEIJJKgMLZnwo1VwH1AkBUxgBRmY8oqLqTjjpPFfWMvyAT87Le/VRQCUjSs3aW+R8BacAYN2ES6RB6DEAeBYDrDPBUyVhBOSM3E0sFCrnUvQCUAYfmbKl4/fHGXNBFGZLEhISjyPVIX8XfEOA5kYMwSuSlRgtd5kbVUYVjvI4IHvZ/GU97w0Hv10mYGTF4htYF4JcQCV0GuaClSCLVWyBEhUfWBgF+QzOW5CWN8maWU8+V/JxE24WbBCqfFIR6gUItHULrOE9np+RoTRjjqxrZDQLpMyr65QFAHqAWmpoYSWiAQzqL3XiNqRhlOo2sJu0ikAPyGTAGG0ZNHtNgrdIoI7hSrupuk29Qz5+TayKJDqd+rpMNVa7DCZgFNSmwZ4q2WDP4avip/yoXI+EchpCu6gcndZSF4n180olSteqj9vb3M0vft8Wu4eC2/zrba4alpLWWldp1xZK0a6fNbydjC3mkLX20Pbbs46PNY62hOX/KGu1GxhXgxJahHidxyZZ6wBBlwMjmDB7eqqq5hXuMrQv4qDqwViPNe60HVG8uf+UKFMUEBtdEAMoGdB4w+KeLGHCe6xLOTtM7dkg1LehBQamr4TdrLOqRysMUP38xWnQrEYSHbas+hgnhwTTrNOcQUpDwk4ml6UXt0o2HmcEI7TgipMQxAfbQfP1aMOay7DItDMCV61kJUXkgsglKzTfG79ukz2Bu/ZKJb548rHp2gnKh1UCNMaoyg6jQoGNU5zG8Ru9Ajpke6oWY2aWW/aXKbuc+YCh5XHUyKKmIxNBfJ62rwV0681VFSvIie6mDVGuGJ8WUubGZVmodEG7tYNVkXKaH7azL4zwjyjn1U7ebE5mbClnxShKX7nnKcwF+HG1r0Lf3n0b96e397/1tlQ81mwrVU5lqvSqRFkS6emeEFq1coEa3K/7ybcgwhAD3VDf9BmSwDtoFFpqB+nviQvQtoFCc0rIrD7fDwChd9nfNzlyzJu88dmAP6FaKqaBosybuMh2wqpOxQiIzk6dDV+7dArznLm6oBFDsrKm2lMKbd+1WXU81tuRlD0lRu5BsS7CNmi1ETOahsvsCgSKikcqe3DmG6G47uCnUytEN4B62R9j73RkjlYw7s2qUlxJsELQg0lBW6AqK5k3gQL1HUYvhOWzWSC/HXfuW55imgHHMUppJ6+LcAbKLA4vfpqCqS1L5wBAXfzK9KmXqjJW1wHdwru1uUsMsV5pa6SJn3F9nytzrJRB/IWtRdhyemiaKp8ZHtwYY4YmSBFNd7/qqpz7iaZSGhhJxiGGM7LteUHgBY56uC+lpjzbi/Eaxj/bCpwxBNlz1TvX7cxmDD4KLSQv6qmXCU50OqtGeO/DwXPpVAgm9xRL4BYLGJM1ZqX9Y4HRAgayXf7X52aCsk0YLfqvDnO+1JN2462y3xplvtvrdFCIMVsCmZa20bWJokkIC7DavnThrxdtehRRQrQg+FfC0r0QaX4rs9loE/JIPCenbm7z/5mvy32k3UL2i0oMLfNkcVqjFmsfZ8lh8VnLo2SvcwJcFozi8/WN2kv9vJG/ZSMpNuy5aiqIapC2XUj+/8uuOy84PX3Xqa6R0RZZrpdc1HoNEzP7rAVU8oYv0KH/aWKyWJB+vay6qvir6oz8c3n/+DstXFVf/cXtXDeRuJENxNK88y9yfPahH2irfj4DZH7niffOCt+GfPba6BgfX1egbVr+tC1wNaLwKBI2lXoCD+nTaxIIdO3hhIGxQrw6dGrQ43Aj7bhvoDriVveL+Mcvn33OvtCsz4/P6hXmHnXoni960nKJ3lj9f2K4Gd6XdrL6T1b0DLd1W2MHqpFzfD8ZjNLjeW5RmnA6u95Q3uv/0W6//FoEp5/4ShzycU5n4ZG+JzHLuKxGWvbcJNIx7ykuH372lBZptT1n2D23MwN3uM0YjwWHl31u+uepmLzXqXxDsveyrWa111DrXw/+Fvf8/dKEWfPV3Pm2MHUkWsKRIIpxZAmZI7AlhseDcKbJq+KJ+t+YvnzIkeE4YG9G/SIZjVVaDc2aSV2TByYUBSIzxuQNf6zkl52zmxpWb/wc).

File: CDBTo3DTiles/src/TileFormatIO.cpp [hunk 4]
```diff
@@ -399,150 +399,191 @@ void createFeatureMetadataClasses(
...
+                // Add propety to feature table
+                metadataExtension["featureTables"][CDB_FEATURE_TABLE_NAME]["class"] = CDB_CLASS_NAME;
+                metadataExtension["featureTables"][CDB_FEATURE_TABLE_NAME]["elementCount"] = instanceCount;
+                metadataExtension["featureTables"][CDB_FEATURE_TABLE_NAME]["properties"][property.first]["bufferView"] = static_cast<int>(gltf->bufferViews.size() - 1);
 
-        tinygltf::Value primitiveExtensionValue;
-        CDBTo3DTiles::ParseJsonAsValue(&primitiveExtensionValue, primitiveExtension);
-        gltf->meshes[0].primitives[0].extensions.insert(std::pair<std::string, tinygltf::Value>(std::string("EXT_feature_metadata"), primitiveExtensionValue));
+            }
+            /*
```
Comment (ErixenCruz) [inline CDBTo3DTiles/src/TileFormatIO.cpp:476]: Should we leave this commented here or delete it?

**Comment by sanjeetsuhag** [comment]
@ErixenCruz This is ready to be merged.

**Pull request merged**